code/game/shuttle_engines.dm

```diff
@@ -109,95 +108,88 @@
 	return TRUE
 
 //Propagates the change to the shuttle.
-/obj/structure/shuttle/engine/proc/alter_engine_power(mod)
+/obj/machinery/power/shuttle_engine/proc/alter_engine_power(mod)
 	if(!mod)
 		return
 	var/obj/docking_port/mobile/port = connected_ship_ref?.resolve()
 	if(port)
 		port.alter_engines(mod)
 
-/obj/structure/shuttle/engine/heater
+/obj/machinery/power/shuttle_engine/heater
 	name = "engine heater"
-	icon_state = "heater"
 	desc = "Directs energy into compressed particles in order to power engines."
+	icon_state = "heater"
+	circuit = /obj/item/circuitboard/machine/engine/heater
 	engine_power = 0 // todo make these into 2x1 parts
 
-/obj/structure/shuttle/engine/platform
-	name = "engine platform"
-	icon_state = "platform"
-	desc = "A platform for engine components."
-	engine_power = 0
-
-/obj/structure/shuttle/engine/propulsion
+/obj/machinery/power/shuttle_engine/propulsion
 	name = "propulsion engine"
 	icon_state = "propulsion"
 	desc = "A standard reliable bluespace engine used by many forms of shuttles."
+	circuit = /obj/item/circuitboard/machine/engine/propulsion
 	opacity = TRUE
 
-/obj/structure/shuttle/engine/propulsion/in_wall
-	name = "in-wall propulsion engine"
-	icon_state = "propulsion_w"
-	density = FALSE
-	opacity = FALSE
-	smoothing_groups = list()
-
-/obj/structure/shuttle/engine/propulsion/left
+/obj/machinery/power/shuttle_engine/propulsion/left
 	name = "left propulsion engine"
 	icon_state = "propulsion_l"
 
-/obj/structure/shuttle/engine/propulsion/right
+/obj/machinery/power/shuttle_engine/propulsion/right
 	name = "right propulsion engine"
 	icon_state = "propulsion_r"
 
-/obj/structure/shuttle/engine/propulsion/burst
+/obj/machinery/power/shuttle_engine/propulsion/burst
 	name = "burst engine"
 	desc = "An engine that releases a large bluespace burst to propel it."
 
-/obj/structure/shuttle/engine/propulsion/burst/cargo
+/obj/machinery/power/shuttle_engine/propulsion/burst/cargo
 	engine_state = ENGINE_UNWRENCHED
 	anchored = FALSE
 
-/obj/structure/shuttle/engine/propulsion/burst/left
+/obj/machinery/power/shuttle_engine/propulsion/burst/left
 	name = "left burst engine"
 	icon_state = "burst_l"
 
-/obj/structure/shuttle/engine/propulsion/burst/right
+/obj/machinery/power/shuttle_engine/propulsion/burst/right
 	name = "right burst engine"
 	icon_state = "burst_r"
 
-/obj/structure/shuttle/engine/router
-	name = "engine router"
-	icon_state = "router"
-	desc = "Redirects around energized particles in engine structures."
+/obj/machinery/power/shuttle_engine/propulsion/in_wall
```

Comment:
In_wall propulsion engines feel distinct enough to me that they shouldn't use the same circuitboard as standard propulsion engines (or at least, if you were to re-assemble it, you would get the full-sized propulsion engine anyways). Right?

Reply:
why do in-wall engines even exist bro i hate this place